pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts

```diff
@@ -0,0 +1,802 @@
+              expect(await authorizer.canPerform(ACTION_1, user, WHERE_2)).to.be.true;
+            });
+
+            it('still can perform the requested actions everywhere', async () => {
+              await authorizer.grantPermission(ACTION_1, user, WHERE_1, { from: getSender() });
+
+              expect(await authorizer.canPerform(ACTION_1, user, EVERYWHERE)).to.be.true;
+              expect(await authorizer.canPerform(ACTION_1, user, WHERE_2)).to.be.true;
+            });
+
+            it('still can perform the requested actions for other contracts', async () => {
+              await authorizer.grantPermission(ACTION_1, user, WHERE_1, { from: getSender() });
+
+              expect(await authorizer.canPerform(ACTION_1, user, NOT_WHERE)).to.be.true;
+              expect(await authorizer.canPerform(ACTION_1, user, WHERE_2)).to.be.true;
+            });
+
+            it('emits an event', async () => {
+              const receipt = await (
+                await authorizer.grantPermission(ACTION_1, user, WHERE_1, { from: getSender() })
+              ).wait();
+              expectEvent.inReceipt(receipt, 'PermissionGranted', {
+                actionId: ACTION_1,
+                account: user.address,
+                where: WHERE_1,
+              });
+            });
+          });
+          context('when granting the permission for everywhere', () => {
+            it('ignores the request and can still perform the requested action everywhere', async () => {
+              await expect(authorizer.grantPermissionGlobally(ACTION_1, user, { from: getSender() })).not.to.be
+                .reverted;
+
+              expect(await authorizer.canPerform(ACTION_1, user, EVERYWHERE)).to.be.true;
+            });
+
+            it('ignores the request and can still perform the requested action in any specific contract', async () => {
+              await expect(authorizer.grantPermissionGlobally(ACTION_1, user, { from: getSender() })).not.to.be
+                .reverted;
+
+              expect(await authorizer.canPerform(ACTION_1, user, NOT_WHERE)).to.be.true;
+              expect(await authorizer.canPerform(ACTION_1, user, WHERE_1)).to.be.true;
+            });
+
+            it('does not emit an event', async () => {
+              const tx = await authorizer.grantPermissionGlobally(ACTION_1, user, { from: getSender() });
+              expectEvent.notEmitted(await tx.wait(), 'PermissionGrantedGlobally');
+            });
+          });
+        });
+      }
+
+      context('when the sender is root', () => {
+        itGrantsPermissionCorrectly(() => root);
+      });
+
+      context('when the sender is granter', () => {
+        sharedBeforeEach('makes a granter', async () => {
+          await authorizer.addGranter(ACTION_1, granter, EVERYWHERE, { from: root });
+          await authorizer.addGranter(ACTION_2, granter, EVERYWHERE, { from: root });
+        });
+        itGrantsPermissionCorrectly(() => granter);
```

Comment:
We could add two `it`s here where `granter` is unable to grant permission over a third action. 

It would also be good to have tests for a non global granter, who is unable to grant permissions over _any_ address. We could maybe make `granter` a granter over `WHERE_1` and `WHERE_2`, so that we don't need to change `itGrantsPermissionCorrectly`, and then simply add a few `it`s.